src/btreemap.rs

```diff
@@ -1856,490 +1946,484 @@ mod test {
-        assert_eq!(
-            child_0.entries(btree.memory()),
-            vec![
-                (b(&[0, 1]), vec![]),
-                (b(&[0, 2]), vec![]),
-                (b(&[0, 3]), vec![]),
-                (b(&[0, 4]), vec![]),
-                (b(&[1, 2]), vec![]),
-            ]
-        );
+            let child_0 = btree.load_node(root.child(0));
+            assert_eq!(child_0.node_type(), NodeType::Leaf);
+            assert_eq!(
+                child_0.entries(btree.memory()),
+                vec![
+                    (b(&[0, 1]), vec![]),
+                    (b(&[0, 2]), vec![]),
+                    (b(&[0, 3]), vec![]),
+                    (b(&[0, 4]), vec![]),
+                    (b(&[1, 2]), vec![]),
+                ]
+            );
 
-        let child_1 = btree.load_node(root.child(1));
-        assert_eq!(child_1.node_type(), NodeType::Leaf);
-        assert_eq!(
-            child_1.entries(btree.memory()),
-            vec![
-                (b(&[1, 6]), vec![]),
-                (b(&[1, 8]), vec![]),
-                (b(&[1, 10]), vec![]),
-                (b(&[2, 1]), vec![]),
-                (b(&[2, 2]), vec![]),
-            ]
-        );
+            let child_1 = btree.load_node(root.child(1));
+            assert_eq!(child_1.node_type(), NodeType::Leaf);
+            assert_eq!(
+                child_1.entries(btree.memory()),
+                vec![
+                    (b(&[1, 6]), vec![]),
+                    (b(&[1, 8]), vec![]),
+                    (b(&[1, 10]), vec![]),
+                    (b(&[2, 1]), vec![]),
+                    (b(&[2, 2]), vec![]),
+                ]
+            );
 
-        let child_2 = btree.load_node(root.child(2));
-        assert_eq!(
-            child_2.entries(btree.memory()),
-            vec![
-                (b(&[2, 4]), vec![]),
-                (b(&[2, 5]), vec![]),
-                (b(&[2, 6]), vec![]),
-                (b(&[2, 7]), vec![]),
-                (b(&[2, 8]), vec![]),
-                (b(&[2, 9]), vec![]),
-            ]
-        );
+            let child_2 = btree.load_node(root.child(2));
+            assert_eq!(
+                child_2.entries(btree.memory()),
+                vec![
+                    (b(&[2, 4]), vec![]),
+                    (b(&[2, 5]), vec![]),
+                    (b(&[2, 6]), vec![]),
+                    (b(&[2, 7]), vec![]),
+                    (b(&[2, 8]), vec![]),
+                    (b(&[2, 9]), vec![]),
+                ]
+            );
 
-        // Tests a prefix that doesn't exist, but is in the middle of the root node.
-        assert_eq!(
-            btree.range(b(&[1, 5])..b(&[1, 6])).collect::<Vec<_>>(),
-            vec![]
-        );
+            // Tests a prefix that doesn't exist, but is in the middle of the root node.
+            assert_eq!(
+                btree.range(b(&[1, 5])..b(&[1, 6])).collect::<Vec<_>>(),
+                vec![]
+            );
```

Comment:
Maybe we can add one more assert here like:
```
assert_eq!(
                btree.range(b(&[1, 5])..b(&[2, 6])).collect::<Vec<_>>(),
                ...
```
With the start of the range which is not in the tree, there are still keys in the tree that fall in the range.